client/landing/gutenboarding/hooks/use-typer.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useInterval } from 'lib/interval/use-interval';
+
+/**
+ * A React hook that returns typing-machine animated strings
+ *
+ * @param {Array<string>} texts An array of strings you want to create the typing effect for
+ * @param {boolean?} enabled Whether the animation is enabled
+ * @param {number?} delayBetweenCharacters The typing delay time between two charactors. Default is 120ms
+ * @param {number?} delayBetweenWords The typing delay time between two words. Default is 1200ms
+ */
+export default function useTyper(
+	texts: Array< string >,
+	enabled = true,
+	delayBetweenCharacters = 120,
+	delayBetweenWords = 1200
+) {
+	const [ charIndex, setCharIndex ] = useState( 0 );
+	const [ wordIndex, setWordIndex ] = useState( 0 );
+	const [ mode, setMode ] = useState( 'TYPING' );
```

Comment:
Shall we restrict this?

```
// Add above hook definition
type TyperMode = 'TYPING' | 'DELETING';
```

```suggestion
	const [ mode, setMode ] = useState< TyperMode >( 'TYPING' );
```

Reply:
Neat. Still not used to TS pampering.